tests/multinode.rs

```diff
@@ -67,30 +39,49 @@ fn converge(
         spy_window,
         spy.sockets.gossip,
         spy.sockets.gossip_send,
-        exit,
+        exit.clone(),
     ).unwrap();
     //wait for the network to converge
     let mut converged = false;
+    let mut rv = vec![];
     for _ in 0..30 {
         let num = spy_ref.read().unwrap().convergence();
-        if num == num_nodes as u64 {
+        let mut v: Vec<ReplicatedData> = spy_ref
+            .read()
+            .unwrap()
+            .table
+            .values()
+            .into_iter()
+            .filter(|x| x.id != me)
+            .filter(|x| x.requests_addr != daddr)
+            .cloned()
+            .collect();
+        if num >= num_nodes as u64 && v.len() >= num_nodes {
+            rv.append(&mut v);
             converged = true;
             break;
         }
         sleep(Duration::new(1, 0));
     }
     assert!(converged);
-    threads.extend(dr.thread_hdls.into_iter());
-    let v: Vec<ReplicatedData> = spy_ref
-        .read()
-        .unwrap()
-        .table
-        .values()
-        .into_iter()
-        .filter(|x| x.id != me)
-        .map(|x| x.clone())
-        .collect();
-    (v.clone(), me)
+    exit.store(true, Ordering::Relaxed);
+    for t in dr.thread_hdls.into_iter() {
```

Comment:
Another variation. :-/

Reply:
we have had 0 bugs in all the variations :)

Reply:
we can fix this in another PR with a synchronous exit on all the server types.